api/models/Centres.js

```diff
@@ -56,61 +56,63 @@ const model = {
     female_cid_name: {
       type: 'array'
     },
-    male_active_movements_in: {
-      collection: 'movement',
-      note: 'this is a workaround until waterline supports conditional joins see balderdashy/waterline#988 and balderdashy/waterline#645',
-      via: 'active_male_centre_in'
-    },
-    male_active_movements_out: {
-      collection: 'movement',
-      note: 'this is a workaround until waterline supports conditional joins see balderdashy/waterline#988 and balderdashy/waterline#645',
-      via: 'active_male_centre_out'
-    },
-    female_active_movements_in: {
-      collection: 'movement',
-      note: 'this is a workaround until waterline supports conditional joins see balderdashy/waterline#988 and balderdashy/waterline#645',
-      via: 'active_female_centre_in'
-    },
-    female_active_movements_out: {
-      collection: 'movement',
-      note: 'this is a workaround until waterline supports conditional joins see balderdashy/waterline#988 and balderdashy/waterline#645',
-      via: 'active_female_centre_out'
-    },
     mo_type: {
       type: 'string',
       required: true,
       defaultsTo: 0
     },
+    events: {
+      collection: 'event',
+      via: 'centre'
+    },
+    detainees: {
+      collection: 'detainee',
+      via: 'centre'
+    },
+    movements: {
+      collection: 'movement',
+      via: 'centre'
+    },
     toJSON: function () {
-      const maleCapacity = this.male_capacity - this.male_in_use;
-      const femaleCapacity = this.female_capacity - this.female_in_use;
+      const unreconciledEventCounter = (gender, operations) => this.unreconciledEvents.reduce((count, event) => {
+        if (event.detainee.gender === gender && operations.indexOf(event.operation) > 0-1) {
```

Comment:
It was to remind me to complain about not being able to write -1 by the eslinting. Feel complained :))